tpot/tpot.py

```diff
@@ -183,10 +180,8 @@ def fit(self, features, classes, feature_names=None):
             training_testing_data = pd.DataFrame(data=features, columns=feature_names)
             training_testing_data['class'] = classes
 
-            new_col_names = {}
-            for column in training_testing_data.columns.values:
-                if type(column) != str:
-                    new_col_names[column] = str(column).zfill(10)
+            new_col_names = {column: str(column).zfill(10) for column
+                in training_testing_data.columns.values if type(column) != str}
```

Comment:
Same readability concern as above wrt list comprehensions. This seems harder to read, and I'm not sure if we gain anything by it?